File: src/main/java/eterea/core/api/rest/model/ReservaComentario.java
/**
 * 
 */
package eterea.core.api.rest.model;

import java.io.Serializable;
import java.time.OffsetDateTime;

import eterea.core.api.rest.kotlin.model.Auditable;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author alma
 *
 */
@Data
@Entity
@Table(name = "reservacomentario")
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class ReservaComentario extends Auditable implements Serializable {/**
	 * 
	 */
	private static final long serialVersionUID = 2388905330353120208L;

	@Id
	@Column(name = "clave")
	private Long reservaComentarioId;
	
	@Column(name = "rec_res_id")
	private Long reservaId;
	
	@Column(name = "rec_timestamp")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime timestamp;
	
	@Column(name = "rec_usuario")
	private String usuario;
	
	@Column(name = "rec_comentario")
	private String comentario;
}

File: src/main/java/eterea/core/service/kotlin/model/ReservaComentario.kt
package eterea.core.service.kotlin.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "reservacomentario")
data class ReservaComentario(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "clave")
    var id: Long? = null,
    @Column(name = "rec_usuario")
    var usuario: String? = null,
    @Column(name = "rec_comentario")
    var comentario: String? = null,
    @ManyToOne
    @JoinColumn(name = "rec_res_id")
    var reserva: Reserva? = null
) : Auditable() {

    class Builder {
        var id: Long? = null
        var usuario: String? = null
        var comentario: String? = null
        var reserva: Reserva? = null

        fun id(id: Long?) = apply { this.id = id }
        fun usuario(usuario: String?) = apply { this.usuario = usuario }
        fun comentario(comentario: String?) = apply { this.comentario = comentario }
        fun reserva(reserva: Reserva?) = apply { this.reserva = reserva }
        fun build() = ReservaComentario(id, usuario, comentario, reserva)
    }
}